skylog/storage/gcp/pubsub/main.go

```diff
@@ -43,13 +45,15 @@ func runSender(ctx context.Context, cli *pubsub.Client) error {
 
 	var results []*pubsub.PublishResult
 
+	sharding := pb.TreeSharding{Levels: uint32(*shardLevels), Shards: uint32(*treeShards)}
+
 	lim := rate.NewLimiter(rate.Limit(*maxRate), *maxJobSize)
 	for index, next := *beginIndex, int64(0); index < *endIndex; index = next {
 		next = index + int64(*maxJobSize)
 		if next > *endIndex {
 			next = *endIndex
 		}
-		job := pb.BuildJob{TreeId: *treeID, Begin: uint64(index), End: uint64(next)}
+		job := pb.BuildJob{TreeId: *treeID, Begin: uint64(index), End: uint64(next), TreeSharding: &sharding}
```

Comment:
Would it be better to make this a ptr from the outset?

Reply:
Done.